pkg/controllers/awsloadbalancercontroller/credentials_request.go

```diff
@@ -0,0 +1,184 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+const (
+	credentialRequestName      = "aws-load-balancer-credentials-request"
+	credentialRequestNamespace = "openshift-cloud-credential-operator"
+)
+
+// currentCredentialsRequest returns true if credentials request exists.
+func (r *AWSLoadBalancerControllerReconciler) currentCredentialsRequest(ctx context.Context, name types.NamespacedName) (bool, *cco.CredentialsRequest, error) {
+	cr := &cco.CredentialsRequest{}
+	if err := r.Client.Get(ctx, name, cr); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+	return true, cr, nil
+}
+
+func (r *AWSLoadBalancerControllerReconciler) ensureCredentialsRequest(ctx context.Context, namespace string, _ *albo.AWSLoadBalancerController) error {
```

Comment:
Why a parameter for the controller instance if we don't use it?

Reply:
Added missing owner reference func call.